packages/yarnpkg-core/sources/Report.ts

```diff
@@ -1,14 +1,23 @@
+import camelcase       from 'camelcase';
 import {PassThrough}   from 'stream';
 import {StringDecoder} from 'string_decoder';
 
+import {EnhancedError} from './EnhancedError';
 import {MessageName}   from './MessageName';
 import {Locator}       from './types';
 
-export class ReportError extends Error {
+export class ReportError extends EnhancedError {
```

Comment:
Why do we need both `ReportError` and `EnhancedError`? Can't we just have `ReportError` accept the extra field? Especially since it already has `reportExtra` which we can use to report extra data on print 🤔